gradle-metric-schema/src/main/java/com/palantir/metric/schema/gradle/GenerateMetricSchemaTask.java

```diff
@@ -82,4 +101,20 @@ private String defaultLibraryName() {
         String rootProjectName = getProject().getRootProject().getName();
         return rootProjectName.replaceAll("-root$", "");
     }
+
+    @Nullable
+    private String defaultLibraryVersion() {
+        // Gradle returns 'unspecified' when there is no version information, which is not orderable.
+        String version = Objects.toString(getProject().getRootProject().getVersion());
+        return OrderableSlsVersion.safeValueOf(version)
+                // Only provide version data for releases and release candidates, not snapshots. This way we
+                // don't invalidate build caches on each commit.
+                // We prefer passing along version information at this level because it will not be mutated
+                // by shading, where the fallback based on 'package.getImplementationVersion()' will reflect
+                // the shadow jar version.
```

Comment:
This makes sense. Honestly, I kinda wish we did this at the `version gitVersion()` level as every time you make a commit you need to rebuild the jar, rebuild any dist/containers containing the jar etc even if not code in the jar changed. Big breaking change though.

Reply:
That would be interesting to try, something like `[most-recent-release][modifier]`, e.g. `1.0.0` -> `1.0.0-snapshot` -> `1.0.0-rc1` -> `1.0.0-rc1-snapshot`/etc.